src/main/java/net/sf/rails/game/specific/_1835/StockRound_1835.java

```diff
@@ -278,5 +275,43 @@ protected void setPriority(String string) {
 			super.setPriority(string);
 		}
 	}
-     
+
+
+    @Override
+    protected void executeShareTransfer(PublicCompany company, List<PublicCertificate> certsToSell, Player dumpedPlayer, int presSharesToSell) {
+
+        BankPortfolio bankTo = (BankPortfolio) pool.getParent();
+
+        if (dumpedPlayer != null && presSharesToSell > 0) {
+            executePresidentTransferAfterDump(company, new TreeSet<>(certsToSell), dumpedPlayer, presSharesToSell, company.getPresident(), bankTo);
+
+            ReportBuffer.add(this, LocalText.getText("IS_NOW_PRES_OF",
+                    dumpedPlayer.getId(),
+                    company.getId()));
+
+        }
+
+        // Transfer the sold certificates
+        Portfolio.moveAll(certsToSell, bankTo);
+    }
+
+    private void executePresidentTransferAfterDump(PublicCompany company, Set<PublicCertificate> certsToSell, Player newPresident, int presSharesToSell, Player oldPresident, BankPortfolio bankTo) {
+        PublicCertificate presidentCert = company.getPresidentsShare();
+
+        SortedSet<PublicCertificate.Combination> newPresidentsReplacementForPresidentShare = CertificatesModel.certificateCombinations(
```

Comment:
`Combination` is unnecessarily qualified with `PublicCertificate`

Reply:
That's true, I'll directly import `Combination`. Thanks!